### PR TITLE
Generate HTML Documentation in Suggestion Entries

### LINKED ISSUE #682: Html documentation in Suggestion Entries.
### Summary
Currently, when we retrieve the suggestion entries from the Engine, we store the documentation as is. Instead, it should be converted to HTML using the docparser.

### Specification
The conversion should be performed when creating `model::suggestion_database::Entry` structure.

### Acceptance Criteria & Test Cases
Unit tests.

### PR DESCRIPTION
**Pull request by BinarySoftware**
### Pull Request Description
closes #682 
Finally, the Documentation Parser and Generator is used somewhere (Yayyy 🎉 🎉 🎉 )
Modifies docstring from suggestion database to HTML code, enabling us to present documentation for nodes in searcher

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [ ] All code conforms to the [Rust](https://github.com/luna/enso/blob/main/docs/style-guide/rust.md).
- [ ] All code has been tested where possible.
- [ ] All code has been profiled where possible.

### CODE REVIEW

File: src/rust/ide/src/model/suggestion_database.rs [hunk 1]
```diff
@@ -11,6 +11,7 @@ use language_server::types::SuggestionDatabaseUpdateEvent;
 pub use language_server::types::SuggestionEntryArgument as Argument;
 pub use language_server::types::SuggestionEntryId as EntryId;
 pub use language_server::types::SuggestionsDatabaseUpdate as Update;
+pub use parser::DocParser as DocParser;
```
Comment (farmaazon) [inline src/rust/ide/src/model/suggestion_database.rs:14]:
```suggestion
use parser::DocParser;
```
And it should be put into right place according to our Rust Style Guide

File: src/rust/ide/src/model/suggestion_database.rs [hunk 2]
```diff
@@ -51,16 +52,18 @@ impl Entry {
         let this = match entry {
             SuggestionEntryAtom {name,module,arguments,return_type,documentation} =>
                 Self {
-                    name,arguments,return_type,documentation,
+                    name,arguments,return_type,
                     module    : module.try_into()?,
                     self_type : None,
+                    documentation: Entry::gen_doc(documentation),
```
Comment (farmaazon) [inline src/rust/ide/src/model/suggestion_database.rs:58]: Align `:`

File: src/rust/ide/src/model/suggestion_database.rs [hunk 3]
```diff
@@ -98,6 +101,21 @@ impl Entry {
     pub fn with_name(self, name:impl Into<String>) -> Self {
         Self {name:name.into(),..self}
     }
+
+    /// Generates HTML documentation for documented suggestion
+    fn gen_doc(doc: Option<String>) -> Option<String> {
```
Comment (farmaazon) [inline src/rust/ide/src/model/suggestion_database.rs:106]: The None actually indicates an error. So this method should return FallibleResult<String>

Then, during conversion we should also log an error when we are unable to parse documentation (and leave `documentation` field as None - as it is now).

File: src/rust/ide/src/model/suggestion_database.rs [hunk 4]
```diff
@@ -98,6 +101,21 @@ impl Entry {
...
+        match doc {
+            Some(d) => {
+                let parser = DocParser::new_or_panic();
+                let output = parser.generate_html_doc_pure(d);
+                match output {
+                    Ok(result) => Some(result),
+                    Err(_)     => None,
+                }
+            },
+            None    => None,
```
Comment (farmaazon) [inline src/rust/ide/src/model/suggestion_database.rs:116]:
```suggestion
            None => None,
```
If lines are not consecutive, then we don't align anytihng.

File: src/rust/ide/src/model/suggestion_database.rs [hunk 5]
```diff
@@ -46,36 +47,46 @@ pub struct Entry {
 
 impl Entry {
     /// Create entry from the structure deserialized from the Language Server responses.
-    pub fn from_ls_entry(entry:language_server::types::SuggestionEntry) -> FallibleResult<Self> {
+    pub fn from_ls_entry(entry:language_server::types::SuggestionEntry, logger:impl AnyLogger)
+        -> FallibleResult<Self> {
         use language_server::types::SuggestionEntry::*;
+        let convert_doc = |doc: Option<String>| doc.and_then(|doc| match Entry::gen_doc(doc) {
+            Ok(d)  => Some(d),
+            Err(err) => {
```
Comment (wdanilo) [inline src/rust/ide/src/model/suggestion_database.rs:55]: alignment

Comment (wdanilo) [inline src/rust/ide/src/model/suggestion_database.rs:55]: there are functions to convert Result to Option, please use them instead, it will be shorter

Comment (BinarySoftware) [reply src/rust/ide/src/model/suggestion_database.rs]: Yes there are - like `.ok()` or `.err()`, but they're unusable in our case, as we want to log the error, which would be impossible with `.ok` and `.err` would lose the value.
Here - all of available transformations : https://docs.google.com/drawings/d/1EOPs0YTONo_FygWbuJGPfikO9Myt5HwtiFUHRuE1JVM/preview

**Pull request merged**